contract-message-predicate/src/predicate_asm.rs

```diff
@@ -0,0 +1,75 @@
+use fuel_asm::{op, RegId};
+
+const GTF_SCRIPT: u16 = 0x00B;
+const GTF_SCRIPT_LEN: u16 = 0x005;
+const GTF_SCRIPT_INPUTS_COUNT: u16 = 0x007;
+const GTF_INPUT_TYPE: u16 = 0x101;
+const GTF_MSG_DATA_LEN: u16 = 0x11A;
+
+const INPUT_MESSAGE_TYPE: u16 = 2;
+const INSTR_PER_WORD: u16 = 2;
+const BYTES_PER_INSTR: u16 = 4;
+
+// Gets the bytecode for the message-to-contract predicate
+pub fn bytecode() -> Vec<u8> {
```

Comment:
This is doable with the Rust SDK out of the box if it would simplify things here a bit.